ripd/rip_nb_state.c

```diff
@@ -107,8 +111,13 @@ const void *ripd_instance_state_neighbors_neighbor_lookup_entry(
 	yang_str2ipv4(args->keys->key[0], &address);
 
 	for (ALL_LIST_ELEMENTS_RO(rip->peer_list, node, peer)) {
-		if (IPV4_ADDR_SAME(&peer->addr, &address))
-			return node;
+		if (args->exact_match) {
+			if (IPV4_ADDR_SAME(&peer->addr, &address))
+				return node;
+		} else {
+			if (IPV4_ADDR_CMP(&peer->addr, &address) >= 0)
+				return node;
```

Comment:
Scary but I guess I can live with it

Reply:
What bothers you here precisely? I think the `args->exact_match` handling in the `lookup_entry()` callbacks is a fair price to pay for operational-data fragmentation support. The commit message of b8379b3965060c4a1da3202ec713a580354c20cc explains why this is necessary.

If you're concerned about the `O(n)` time complexity of iterating over the linked list until finding the next item, then yeah, that's not cool. This should however be easy to address by changing `rip->peer_list` to be an RB-tree (something that can be done later).